redisvl/llmcache/semantic.py

```diff
@@ -114,6 +114,10 @@ def set_threshold(self, threshold: float):
             raise ValueError("Threshold must be between 0 and 1.")
         self._threshold = float(threshold)
 
+    def clear(self):
+        """Clear the LLMCache and create a new underlying index."""
+        self._index.delete(drop=True)
```

Comment:
So this is one area to think more about... Ideally I would like to allow clearing out the data itself, but not the index. The manual way is to search for all data in the index, get list of keys, and delete them. But this is messy.

Alternatively, we could use the delete method here along with a construction of a new index. But, this gave me some issues with creating an index again with the right params/attributes. Maybe you can shed light here.

Reply:
My guess would be a ``scan_iter`` across the DB using fuzzy match and then delete from index. This is similar to the feast update approach. Don't love it, but nothing else is coming to mind currently.

Reply:
Got it - yeah it won't be too bad. Will implement now and push commit for review along with a test